rotkehlchen/chain/evm/decoding/velodrome/balances.py

```diff
@@ -34,6 +52,63 @@ def __init__(
             deposit_event_types={(HistoryEventType.DEPOSIT, HistoryEventSubType.DEPOSIT_FOR_WRAPPED)},  # noqa: E501
             gauge_deposit_event_types={(HistoryEventType.DEPOSIT, HistoryEventSubType.DEPOSIT_ASSET)},  # noqa: E501
         )
+        self.protocol_token = protocol_token
+        self.voting_escrow_address = voting_escrow_address
 
     def get_gauge_address(self, event: 'EvmEvent') -> ChecksumEvmAddress | None:
         return event.address
+
+    def query_balances(self) -> BalancesSheetType:
+        balances = super().query_balances()
+        if len(addresses_with_deposits := self.addresses_with_activity(event_types=self.deposit_event_types)) == 0:  # noqa: E501
+            return balances
+
+        addresses_to_token_ids = {}
+        for address, events in addresses_with_deposits.items():
+            addresses_to_token_ids[address] = [event.extra_data['token_id'] for event in events if event.extra_data is not None]  # noqa: E501
+
+        voting_escrow_contract = EvmContract(
+            address=self.voting_escrow_address,
+            abi=VOTING_ESCROW_ABI,
+            deployed_block=0,
+        )
+        chunk_size, call_order = get_chunk_size_call_order(self.evm_inquirer)
+        for user_address, token_ids in addresses_to_token_ids.items():
+            if not (results := self.evm_inquirer.multicall(
+                calls=[
+                    (
+                        voting_escrow_contract.address,
+                        voting_escrow_contract.encode(method_name='locked', arguments=[token_id]),
+                    )
+                    for token_id in token_ids
+                ],
+                call_order=call_order,
+                calls_chunk_size=chunk_size,
+            )):
+                log.error(f'Failed to query {self.counterparty} locked balances for address {user_address}')  # noqa: E501
+                continue
+
+            for idx, result in enumerate(results):
+                balance, _, _ = voting_escrow_contract.decode(
+                    result=result,
+                    method_name='locked',
+                    arguments=[token_ids[idx]],
+                )[0]
+                if balance == 0:
+                    continue
+
+                if (price := Inquirer().find_usd_price(self.protocol_token)) == ZERO_PRICE:
+                    log.error(
+                        f'Failed to request the USD price of {self.protocol_token.evm_address}. '
+                        f"{self.counterparty} locked balances won't be accurate.",
+                    )
+
+                balances[user_address].assets[self.protocol_token] = Balance(
```

Comment:
```suggestion
                balances[user_address].assets[self.protocol_token] += Balance(
```
Shouldn't this be adding to the existing balance, not replacing it? maybe I'm misunderstanding how this logic works, but if there are multiple token_ids it looks like it would only get the balance for the last one as the code is currently.